WordPress/Classes/ViewRelated/Tools/SettingsTextViewController.h

```diff
@@ -1,13 +1,31 @@
 #import <UIKit/UIKit.h>
 
+
+typedef void (^SettingsTextChanged)(NSString *);
+
+// Reusable component that renders a UITextField + Hint onscreen. Useful for Text / Password / Email data entry.
+//
 @interface SettingsTextViewController : UITableViewController
 
-@property (nonatomic, copy) void(^onValueChanged)(NSString *);
-@property (nonatomic, copy) void(^onCancel)();
+// Block to be executed on dismiss, if the value was effectively updated.
+//
+@property (nonatomic, copy) SettingsTextChanged onValueChanged;
+
+// Property to indicate whether email validation should be performed, or not.
+//
+@property (nonatomic, assign) BOOL isEmail;
```

Comment:
A quick example of what I meant:

https://gist.github.com/koke/615c07e07c4ec08e86e5edaea275c190